main.cpp

```diff
@@ -1054,11 +1054,11 @@ int main(int argc, char ** argv) {
                     embd_inp.insert(embd_inp.end(), inp_sfx.begin(), inp_sfx.end());
                 }
 
-                remaining_tokens -= line_inp.size();
+                remaining_tokens = params.n_predict - line_inp.size();
```

Comment:
This can get bigger then remaining space in context.
and after https://github.com/ggerganov/llama.cpp/blob/da5303c1ea68aa19db829c634f1e10d08d409680/main.cpp#L850 remaining_tokens, are actually all the space that is left. no?

Reply:
resetting remaining_tokens to params.n_predict would only make sense when we reset the memory, which we don't right now. see https://github.com/ggerganov/llama.cpp/issues/71